xrpl/models/transactions/vault_create.py

```diff
@@ -0,0 +1,110 @@
+"""Represents a VaultCreate transaction on the XRP Ledger."""
+
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Dict, Optional, Union
+
+from typing_extensions import Self
+
+from xrpl.models.amounts.mpt_amount import MPTIssue
+from xrpl.models.currencies import Currency
+from xrpl.models.flags import FlagInterface
+from xrpl.models.required import REQUIRED
+from xrpl.models.transactions.transaction import Transaction
+from xrpl.models.transactions.types import TransactionType
+from xrpl.models.utils import KW_ONLY_DATACLASS, require_kwargs_on_init
+
+_MAX_DATA_LENGTH = 256 * 2
+_MAX_MPTOKEN_METADATA_LENGTH = 1024 * 2
+_MAX_DOMAIN_ID_LENGTH = 32 * 2
+
+
+class VaultCreateFlag(int, Enum):
+    """Flags for the VaultCreate transaction."""
+
+    TF_VAULT_PRIVATE = 0x00010000
+    """
+    Indicates that the vault is private. It can only be set during Vault creation.
+    """
+    TF_VAULT_SHARE_NON_TRANSFERABLE = 0x00020000
+    """
+    Indicates the vault share is non-transferable. It can only be set during Vault
+    creation.
+    """
+
+
+class VaultCreateFlagInterface(FlagInterface):
+    """Interface for the VaultCreate transaction flags."""
+
+    TF_VAULT_PRIVATE: bool
+    """
+    Indicates that the vault is private. It can only be set during Vault creation.
+    """
+    TF_VAULT_SHARE_NON_TRANSFERABLE: bool
+    """
+    Indicates the vault share is non-transferable. It can only be set during Vault
+    creation.
+    """
+
+
+class WithdrawalPolicy(int, Enum):
+    """Withdrawal policy for the Vault."""
+
+    VAULT_STRATEGY_FIRST_COME_FIRST_SERVE = 1
+    """Requests are processed on a first-come-first-serve basis."""
+
+
+@require_kwargs_on_init
+@dataclass(frozen=True, **KW_ONLY_DATACLASS)
+class VaultCreate(Transaction):
+    """The VaultCreate transaction creates a new Vault object."""
+
+    asset: Union[Currency, MPTIssue] = REQUIRED  # type: ignore
```

Comment:
```suggestion
    asset: Union[Currency] = REQUIRED  # type: ignore
```
This can be delete since `MPTCurrency` is covered in `Currency`.